src/Catalog.API/Extensions/Extensions.cs

```diff
@@ -46,7 +46,7 @@ public static void AddApplicationServices(this IHostApplicationBuilder builder)
         else if (!string.IsNullOrWhiteSpace(builder.Configuration.GetConnectionString("openai")))
         {
             builder.AddOpenAIClientFromConfiguration("openai");
-            builder.Services.AddEmbeddingGenerator(sp => sp.GetRequiredService<OpenAIClient>().AsEmbeddingGenerator(builder.Configuration["AI:OpenAI:EmbeddingModel"]!))
+            builder.Services.AddEmbeddingGenerator(sp => sp.GetRequiredService<OpenAIClient>().GetEmbeddingClient(builder.Configuration["AI:OpenAI:EmbeddingModel"]!).AsIEmbeddingGenerator())
```

Comment:
This should be changed, we pass the model in the connection string and this should not be needed anymore.

cc @eerhardt

Reply:
Fixed in the latest code.